tests/Feature/GroupBalancedDivisionTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+use App\Helpers\GroupBalancedDivision;
+use App\Models\Event;
+use App\Models\User;
+use App\Models\Course;
+use App\Models\Registration;
+use App\Models\Group;
+
+
+uses(\Illuminate\Foundation\Testing\RefreshDatabase::class);
+
+test('equal div only drinkers', function () {
+    $event = Event::factory()->create(['consider_alcohol' => true])->first();
+    $courses = Course::factory()->count(10)->create();
+    $groups = Group::factory()->count(10)->create(['event_id' => $event->id]);
+
+    foreach ($courses as $course) {
+        User::factory()->count(20)->create(['course_id' => $course->id]);
+    }
+
+    foreach (User::all() as $user) {
+        Registration::factory()->create([
+            'event_id' => $event->id,
+            'user_id' => $user->id,
+            'drinks_alcohol' => true,
+            'fulfils_requirements' => true,
+            'queue_position' => null
+        ]);
+    }
+
+    $div = new GroupBalancedDivision($event, true);
+    $div->assign();
+
+    // each group should contain 2 users per course
+    foreach ($groups as $group) {
+        foreach ($courses as $course) {
+            expect($group->registrations()
+                ->join('users', 'registrations.user_id', '=', 'users.id')
+                ->where('users.course_id', '=', $course->id)
+                ->count())->toEqual(2);
+        }
+    }
+});
+
+test('unequal div only drinkers', function () {
```

Comment:
The only thing that is uequal here in the test is that there are always 10 more users per degree programme. They are created again in chronological order. What added value does this test offer compared to the previous one?

Reply:
rewrote test